### PR TITLE
Please merge my cleanup

### PR DESCRIPTION
**Pull request by cr**
Running the genmd.py script revealed a number of XHTML issues and broken links in the existing advisory files. The first commit of this PR fixes those. The second commit addresses a bug in and some improvements to genmd.py itself.

### CODE REVIEW

**Comment by pmclanahan** [comment]
@albill the changes to the markdown look okay to you? Seems fine to me. I'm looking at the `genmd.py` changes.

File: genmd.py [hunk 1]
```diff
@@ -290,6 +291,7 @@ def parser(txt):
                 raise Exception("Unknown MFSA header: %s" % line)
 
         # body = minidom.parseString("<html>" + bod + "</html>")
+        # print MfsaMd.xmlheader + bod + MfsaMd.xmlfooter
```
Comment (pmclanahan) [inline genmd.py]: Should probably just delete this line.

**Comment by pmclanahan** [comment]
Changes look fine to me. I'd like @albill to give his r+ before merge. Thanks!

File: genmd.py (hunk 1 above)
Comment (cr) [inline genmd.py]: I kept the line around because I regularly need it for debugging and I'm too lazy to write proper debug output filtering.

**Comment by albill** [comment]
R+

**Pull request merged**